### ISSUE TITLE
testng ignore cases packed in jar when execute via groups?

### ISSUE THREAD

**Issue by zutuanwang** [cbeust/testng#1451]
### TestNG Version
> Note: 6.9.9

### Expected behavior
I pack some cases in jar, and put them in classpath of another project, and then try to execute via group, and found no cases has executed. Are they ignore?

### Is the issue reproductible on runner?

- [ ] Shell
- [ ] Maven
- [ ] Gradle
- [ ] Ant
- [x] Eclipse
- [ ] IntelliJ
- [ ] NetBeans

### Test case sample
(they are in jar)

```java
public class TestNgSample {
	@Test(groups={"all"})
	public void testTrue() {
		System.out.println("...");
	}
}

public class TestNgSample2 {
	@Test(groups={"all"})
	public void testTrue() {
		System.out.println("...");
	}
}
```

Try to cases of group "all" in another project, but none executed.

**Comment by krmahadevan** [comment]
@zutuanwang - How are you creating your jar ? Is it using a build tool such as Maven or gradle ? How does your project structure look like ? 
Since these information would help us figure out what is going wrong, can you please help do the following ?

Create a sample project on GitHub which we can clone and build into a jar to reproduce the problem. That way its pretty easy to figure out what is going wrong and help you out.

**Comment by zutuanwang** [comment]
Thanks for feedback.

The demo project address is:
https://github.com/zutuanwang/TestProject2.git
or
git@github.com:zutuanwang/TestProject2.git
As below, testcases were packed in runCases.jar.
![image](https://user-images.githubusercontent.com/13378780/27215423-b9a1d392-52a3-11e7-8c34-3b4ee82244ee.png)

and in MoreGroupsExecutor try to run them via groups(all, function), but none case executed.
below is the output message.
{code:xml}
Request params: {Groups:[{GroupName:'function'},{GroupName:'all'}],Packes:[{PackName:'tc.dm.*'},{PackName:'tc.yl.*'}]}
Begin to run test
...
... TestNG 6.9.9 by Cédric Beust (cedric@beust.com)
...

[TestRunner] Running the tests in 'TestMoreGroups: cases in jar' with parallel mode:false
[RunInfo] Adding method selector: org.testng.internal.XmlMethodSelector@48533e64 priority: 10
[TestNG] Running:
  Command line suite

[SuiteRunner] Created 1 TestRunners
[TestRunner] Running test TestMoreGroups: cases in jar on 0  classes,  included groups:[all function ] excluded groups:[]
===== Invoked methods
=====
Creating F:\java_spaces\jse\TestProject2\test-output\MoreGroupsSuite\TestMoreGroups_ cases in jar.html
Creating F:\java_spaces\jse\TestProject2\test-output\MoreGroupsSuite\TestMoreGroups_ cases in jar.xml

===============================================
    TestMoreGroups: cases in jar
    Tests run: 0, Failures: 0, Skips: 0
===============================================
{code}

**Comment by krmahadevan** [comment]
Tried reproducing this issue with latest released version of TestNG `7.5`. This is no longer a problem.

[This project](https://github.com/cbeust/testng/files/7829938/issue_1451.zip) can be used to cross check this.
